### ISSUE TITLE
Failed to merge directives

### ISSUE THREAD

**Issue by MarcosBustamante** [mununki/gqlmerge#33]
I am using entgo and the framework generate and apply directives on the schema file. The gqlmerge is failing to merge schemas with these directives.

schema.graphql
```
directive @goModel(
  model: String
  models: [String!]
) on OBJECT | INPUT_OBJECT | SCALAR | ENUM | INTERFACE | UNION

enum Color @goModel(model: "backend/ent/color.Color") {
  Blue
  Red
}
```

Error
```
🎉 [graph/auth] Total 6 GraphQL files found!
panic: /Users/x/Documents/Projects/www/backend/graph/auth/register.mutation.graphql:1:11: unexpected "@", expected Ident

goroutine 1 [running]:
github.com/mattdamon108/gqlmerge/lib.(*Lexer).ConsumeToken(0x14000100280, 0xfffffffe)
        /Users/x/go/pkg/mod/github.com/mattdamon108/gqlmerge@v0.2.5/lib/lexer.go:104 +0x16c
github.com/mattdamon108/gqlmerge/lib.(*Lexer).ConsumeIdent(0x14000100280)
        /Users/x/go/pkg/mod/github.com/mattdamon108/gqlmerge@v0.2.5/lib/lexer.go:88 +0x3c
github.com/mattdamon108/gqlmerge/lib.(*Schema).ParseSchema(0x140001060f0, 0x14000100280)
        /Users/x/go/pkg/mod/github.com/mattdamon108/gqlmerge@v0.2.5/lib/parse.go:15 +0x44
github.com/mattdamon108/gqlmerge/lib.getSchema({0x16d53b7ce?, 0xa?})
        /Users/x/go/pkg/mod/github.com/mattdamon108/gqlmerge@v0.2.5/lib/merge.go:50 +0xd0
github.com/mattdamon108/gqlmerge/lib.Merge({0x1400001406c, 0x4}, {0x140000101c0, 0x1, 0x1400004a6b8?})
        /Users/x/go/pkg/mod/github.com/mattdamon108/gqlmerge@v0.2.5/lib/merge.go:18 +0xa0
main.main()
        /Users/x/go/pkg/mod/github.com/mattdamon108/gqlmerge@v0.2.5/main.go:21 +0xbc
exit status 2
```

**Comment by mununki** [comment]
Thanks for opening the issue, it looks like the parser isn't handling the directive, I'll try to find some time to fix it.

**Comment by MarcosBustamante** [comment]
Thank you for taking a look on it @mununki. Since you are going to review directives declaration and the usage on type level, it would be nice to check it at field level too. This [document](https://github.com/nautilus/gateway/blob/master/docs/mergingStrategies.md) also share some merge strategies that might help you testing.

**Comment by mununki** [comment]
By any chance, can you provide a reference to the GraphQL spec documentation for the directives shown in the sample code above?

**Comment by MarcosBustamante** [comment]
Sure. The [entgo](https://entgo.io/docs/tutorial-todo-gql/#running-schema-generation) doesn't provide a deep explanation about directives because it use gqlgen to generate the GraphQL schema. 

On the [gqlgen documentation](https://gqlgen.com/reference/directives/) you are able to find a better explanation about directives.

Finally, the [GraphQL documentation](https://graphql.org/learn/queries/#directives) brings all the information needed to understand this concept.

**Comment by shyaaam** [comment]
I face the same issue.

**Comment by mununki** [comment]
~ref: http://spec.graphql.org/October2021/#Directives~
https://the-guild.dev/graphql/tools/docs/schema-directives

**Comment by mununki** [comment]
But only consider schema directives

**Comment by MarcosBustamante** [comment]
Hello @mununki, do you think that this problem can be fixed? Do you need more context around directives?

**Comment by mununki** [comment]
> Hello @mununki, do you think that this problem can be fixed? Do you need more context around directives?

I think I've gathered the necessary context. I'll try to fix it as soon as possible.

**Comment by mununki** [comment]
@MarcosBustamante @shyaaam I fixed and released v0.2.6 to support directives. Can you test it?

Side comment: I also release to the homebrew, but it seems not working properly. I failed to install with brew with this error
```
❯ brew install mununki/tools/gqlmerge
Error: mununki/tools/gqlmerge: wrong number of arguments (given 1, expected 0)
```

**Comment by mununki** [comment]
Now brew formula is fixed. You can download it with brew.
```
$ brew untap mununki/tools
$ brew install mununki/tools/gqlmerge
```

**Comment by MarcosBustamante** [comment]
I tested running through go run and it worked perfectly. Thank you for the quick fix

**Comment by MarcosBustamante** [comment]
@mununki It is failing when the directive has multiple arguments
```
directive @goField(forceResolver: Boolean, name: String) on FIELD_DEFINITION | INPUT_FIELD_DEFINITION
```
Error:
```
🎉 [graph/platform/schema.graphql] Total 1 GraphQL files found!
panic: /Users/bustamante/Documents/Projects/www/backend/graph/platform/schema.graphql:1:42: unexpected ",", expected Ident

goroutine 1 [running]:
github.com/mununki/gqlmerge/lib.(*Lexer).ConsumeToken(0x14000010180, 0xfffffffe)
        /Users/bustamante/go/pkg/mod/github.com/mununki/gqlmerge@v0.2.6/lib/lexer.go:103 +0x16c
github.com/mununki/gqlmerge/lib.(*Lexer).ConsumeIdent(0x14000010180)
        /Users/bustamante/go/pkg/mod/github.com/mununki/gqlmerge@v0.2.6/lib/lexer.go:81 +0x3c
github.com/mununki/gqlmerge/lib.ParseArgument(0x14000010180)
        /Users/bustamante/go/pkg/mod/github.com/mununki/gqlmerge@v0.2.6/lib/parse.go:781 +0xd8
github.com/mununki/gqlmerge/lib.(*Schema).ParseSchema(0x14000106120, 0x14000010180)
        /Users/bustamante/go/pkg/mod/github.com/mununki/gqlmerge@v0.2.6/lib/parse.go:36 +0x840
github.com/mununki/gqlmerge/lib.getSchema({0x16bb53809?, 0x0?})
        /Users/bustamante/go/pkg/mod/github.com/mununki/gqlmerge@v0.2.6/lib/merge.go:50 +0xd0
github.com/mununki/gqlmerge/lib.Merge({0x1400001406c, 0x2}, {0x140000100a0, 0x1, 0x1400004a6b8?})
        /Users/bustamante/go/pkg/mod/github.com/mununki/gqlmerge@v0.2.6/lib/merge.go:18 +0xa0
main.main()
        /Users/bustamante/go/pkg/mod/github.com/mununki/gqlmerge@v0.2.6/main.go:21 +0xbc
exit status 2
```

**Comment by mununki** [comment]
Ah didn't expect there's comma between arguments. Looks easy to fix.

**Comment by mununki** [comment]
Can you test with v0.2.7? https://github.com/mununki/gqlmerge/releases/tag/v0.2.7

Seems fine for my test cases.

**Comment by mununki** [comment]
Fixed some regarding parsing arguments in application of directives. Can you test with v0.2.8?

**Comment by mununki** [comment]
Hopefully the issue is fixed in the latest version, v0.2.10. I'll close the issue for now, but if you need to reopen it, please do so.
